poll/poll.py

```diff
@@ -904,8 +907,8 @@ def student_view(self, context=None):
             'can_view_private_results': self.can_view_private_results(),
             # a11y: Transfer block ID to enable creating unique ids for questions and answers in the template
             'block_id': self._get_block_id(),
+            'usage_id': six.text_type(self.scope_ids.usage_id),
```

Comment:
Why do we want to ignore `self.location.html_id()`, which can be returned by `_get_block_id()`?

Reply:
Okay so what is happening is that one of the test (https://github.com/open-craft/xblock-poll/blob/master/tests/integration/test_functions.py#L269) was failing because it required template to generate unique radio buttons for each survey provided there were more than one survey. Previously it was using value of _get_block_id() to create that and during template rendering each time the function gets a unique value. But with a change in xblock-sdk code it seems that _get_block_id() is not giving unique ids (now it returns course-v1:edX+Demo+2020) during each survey rendering. Inspecting code takes one to xblock-sdk code from where the rendering (https://github.com/edx/xblock-sdk/commit/c4868e0de15d748a79f47fff3bf3cf217f87c318#diff-a6fda0d2b24a612bffcb978acbbc76e2R314) is taking place and this line (https://github.com/edx/xblock-sdk/commit/c4868e0de15d748a79f47fff3bf3cf217f87c318#diff-a6fda0d2b24a612bffcb978acbbc76e2R301) is suggesting that the value is being returned as a Mock.   

Previous values used is now populated in self._get_block_id() so iam now using it to achieve uniqueness and pass the test.

Reply:
Got it, thank you for explaining this.